src/hidimstat/desparsified_lasso.py

```diff
@@ -4,141 +4,108 @@
     verbose=0,
+    group=False,
+    cov=None,
+    noise_method="AR",
+    order=1,
+    stationary=True,
 ):
-    """Desparsified Lasso with confidence intervals
+    """
+    Desparsified Lasso
+
+    Algorithm based on Algorithm 1 of d-Lasso and d-MTLasso in
+    :cite:`chevalier2020statistical`.
 
     Parameters
     ----------
     X : ndarray, shape (n_samples, n_features)
-        Data.
+        Input data matrix.
 
-    y : ndarray, shape (n_samples,)
-        Target.
+    y : ndarray, shape (n_samples,) or (n_samples, n_times)
+        Target vector for single response or matrix for multiple
+        responses.
 
     dof_ajdustement : bool, optional (default=False)
-        If True, makes the degrees of freedom adjustement (cf. [4]_ and [5]_).
-        Otherwise, the original Desparsified Lasso estimator is computed
-        (cf. [1]_ and [2]_ and [3]_).
-
-    confidence : float, optional (default=0.95)
-        Confidence level used to compute the confidence intervals.
-        Each value should be in the range [0, 1].
+        If True, applies degrees of freedom adjustment from :footcite:t:`bellec2022biasing`.
+        If False, computes original Desparsified Lasso estimator.
 
     max_iter : int, optional (default=5000)
-        The maximum number of iterations when regressing, by Lasso,
-        each column of the design matrix against the others.
+        Maximum iterations for Nodewise Lasso regressions.
 
     tol : float, optional (default=1e-3)
-        The tolerance for the optimization of the Lasso problems: if the
-        updates are smaller than `tol`, the optimization code checks the
-        dual gap for optimality and continues until it is smaller than `tol`.
-
-    residual_method : str, optional (default='lasso')
-        Method used for computing the residuals of the Nodewise Lasso.
-        Currently the only method available is 'lasso'.
+        Convergence tolerance for optimization.
 
     alpha_max_fraction : float, optional (default=0.01)
-        Only used if method='lasso'.
-        Then alpha = alpha_max_fraction * alpha_max.
+        Fraction of max alpha used for Lasso regularization.
 
-    n_jobs : int or None, optional (default=1)
-        Number of CPUs to use during the Nodewise Lasso.
+    eps : float, optional (default=1e-2)
+        Small constant used in noise estimation.
+
+    tol_reid : float, optional (default=1e-4)
+        Tolerance for variance estimation with the Reid method.
+
+    n_split : int, optional (default=5)
+        Number of splits for cross-validation in Reid procedure.
+
+    n_jobs : int, optional (default=1)
+        Number of parallel jobs. Use -1 for all CPUs.
+
+    seed : int, default=0
+        Random seed for reproducibility.
+
+    verbose : int, default=0
+        Verbosity level for logging.
+
+    group : bool, default=False
+        If True, use group Lasso for multiple responses.
 
-    memory : str or joblib.Memory object, optional (default=None)
```

Comment:
Please do not remove that.

Reply:
Which function do you want to cache in desparsified lasso?

Reply:
I was missing the next comment. I cache the function _compute_all_residuals and at the same time, I cache the function reid.